alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/settings/DestinationMailSettings.kt

```diff
@@ -0,0 +1,68 @@
+package com.amazon.opendistroforelasticsearch.alerting.settings
+
+import org.elasticsearch.common.settings.Setting
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.settings.SecureSetting
+import org.elasticsearch.common.settings.SecureString
+import java.io.IOException
+
+/**
+ * settings specific to mail destination.
+ */
+data class DestinationMailSettings(
+    val host: String,
+    val port: Int,
+    val method: String,
+    val from: String,
+    val username: SecureString,
+    val password: SecureString
+) {
+    companion object {
+        val DESTINATION_MAIL_HOST = Setting.simpleString(
+                "opendistro.alerting.destination.mail.host",
+                "localhost",
+                Setting.Property.NodeScope
+        )
+
+        val DESTINATION_MAIL_PORT = Setting.intSetting(
+                "opendistro.alerting.destination.mail.port",
+                587,
+                Setting.Property.NodeScope
+        )
+
+        val DESTINATION_MAIL_METHOD = Setting.simpleString(
+                "opendistro.alerting.destination.mail.method",
+                "none",
+                Setting.Property.NodeScope
+        )
+
+        val DESTINATION_MAIL_FROM = Setting.simpleString(
+                "opendistro.alerting.destination.mail.from",
+                "opendistro-alerting@localhost",
+                Setting.Property.NodeScope
+        )
+
+        val DESTINATION_MAIL_USERNAME = SecureSetting.secureString(
```

Comment:
I realized after my initial review of the PR that these keystore settings are not reloadable by default and as such would require node restarts to update. This could be a troublesome user experience so I looked into how to make them reloadable.

[Here](https://github.com/qreshi/alerting/commit/3388802bb79a058cd69400f42a4931b6328a0d0f) is an example of some changes I made locally and tested which allowed the keystore settings to be updated while the ES process is running by calling the following reload API after changing the respective keystore value:

```
curl -X POST "localhost:9200/_nodes/reload_secure_settings?pretty" -H 'Content-Type: application/json' -d'
{
  "secure_settings_password": "keystore_password" 
}
'
```

Also if these settings become reloadable, we would need the settings above to be dynamic and mutable at runtime as well then. The suggestion to changing `DestinationSettings` that I'm looking into should help address this.